lib/chai/core/assertions.js

```diff
@@ -1402,7 +1402,8 @@ module.exports = function (chai, _) {
       }
     }
 
-    if (errMsgMatcher) {
+    if (errMsgMatcher !== undefined && errMsgMatcher !== null &&
+        caughtErr !== undefined && caughtErr !== null) {
```

Comment:
What you said about `caughtErr` makes sense, I did it because I thought of handling throwing empty strings at first, but then changed my mind and thought about asking what you think before handling anything other than `Error` objects.

Hmmm, I don't feel so confortable using loose equality comparisons, IMO having some more characters to explicitly check for `null` and `undefined` makes things more explicit, even though the effect is the same in the end. What do you think? If you guys still prefer using loose equality we can do it though 😄